pytests/tests/test_rpm_cmdline.py

```diff
@@ -156,3 +156,20 @@ def test_install_same_as_filname(utils):
     ret = utils.run(['tdnf', 'install', '-y', '--nogpgcheck', pkgname])
     assert(ret['retval'] == 0)
     assert(utils.check_package(pkgname))
+
+
+# test "tdnf reinstall /path/to/pkg.rpm"
+def test_reinstall_as_file(utils):
+    pkgname = utils.config["sglversion_pkgname"]
+    path = get_pkg_file_path(utils, pkgname)
+
+    # prepare by installing package
+    ret = utils.run(['tdnf', 'install', '-y', '--nogpgcheck', path])
+    assert(ret['retval'] == 0)
+    assert(utils.check_package(pkgname))
+
+    # actual test
+    ret = utils.run(['tdnf', 'reinstall', '-y', '--nogpgcheck', path])
+    assert(ret['retval'] == 0)
+    assert(utils.check_package(pkgname))
+    assert("Nothing to do" not in ret['stderr'])
```

Comment:
also check for "Installing ..." string in stdout. One more verification to enure it got installed again